tests/net/lib/http_server/json/src/main.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "server_functions.h"
+
+#include <zephyr/net/http/service.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/ztest.h>
+
+#define BUFFER_SIZE  256
+#define STACK_SIZE   8192
+#define MY_IPV4_ADDR "127.0.0.1"
+#define SERVER_PORT  8000
+#define TIMEOUT      1000
+
+static struct k_sem server_sem;
+
+static K_THREAD_STACK_DEFINE(server_stack, STACK_SIZE);
```

Comment:
The HTTP Server should define its own thread stack, I would assume.

Really, the application should just be defining resources, metadata, and potentially custom handlers.